packages/framework-types/src/concepts/role.ts

```diff
@@ -14,6 +17,12 @@ export interface AuthMetadata {
   readonly skipConfirmation?: boolean
 }
 
+export type BeforeFunction = (
+  filter: FilterFor<Class<ReadModelInterface>>,
+  currentUser?: UserEnvelope
+) => FilterFor<Class<ReadModelInterface>>
+
 export interface RoleAccess {
   readonly authorize: 'all' | Array<Class<RoleInterface>>
+  readonly before?: Array<BeforeFunction>
```

Comment:
The same with this field, I don't think it belongs to this interface. 
I would create an interface called `FilterHooks` (for example) with just this `before` field (for now). That type could be defined in the same way the decorator `@ReadModel` is defined (in the core package).

Then, in the decorator, you can say that the parameter conforms both interfaces, like this:
```typescript
export function ReadModel(attributes: RoleAccess & FilterHooks): (readModelClass: Class<ReadModelInterface>) => void {
...
```
What do you think?

Reply:
Definitely, it has nothing to do with roles now that I think about it lol